packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -24,6 +24,70 @@ export interface PreconditionChecker {
   check(): boolean;
 }
 
+export interface PostconditionChecker<T> {
```

Comment:
Added a PostconditionChecker interface that will check inputs gathered by ParameterGatherers. If it passes then the inputs will just be passed along, otherwise it will be replaced by a CancelResponse.